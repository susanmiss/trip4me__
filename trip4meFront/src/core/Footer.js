import React from "react";
import { Link } from 'react-router-dom'

let year = new Date();

const Footer = () => (

  <div>
    <footer className="footer">
      <div className="container">
        <div className="row align-items-center">
          <div className="col-md-4">
            <span className="copyright">Copyright &copy; Trip4me {year.getFullYear()}</span>
          </div>
          <div className="col-md-4">
            <ul className="list-inline social-buttons">
              <li className="list-inline-item">
                <a href="https://www.instagram.com/susan_missaglia" target="_blank">
                  <i className="fab fa-instagram"></i>
                </a>
              </li>
              <li className="list-inline-item">
                <a href="https://www.github.com/susanmiss" target="_blank">
                  <i className="fab fa-github"></i>
                </a>
              </li>
              <li className="list-inline-item">
                <a href="https://www.linkedin.com/in/susan-missaglia/" target="_blank">
                  <i className="fab fa-linkedin-in"></i>
                </a>
              </li>
            </ul>
          </div>
          <div className="col-md-4">
            <ul className="list-inline quicklinks">
              <li className="list-inline-item">
                <a href="http://www.susanmissaglia.co.uk" target="_blank">By Susan Missaglia</a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </footer>

  </div>
);

export default Footer;
