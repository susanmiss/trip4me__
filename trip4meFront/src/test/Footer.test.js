import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Footer from '../core/Footer'


const setup = (props={}, state=null) => {
    const wrapper = shallow(<Footer {...props} />)
    if( state ) wrapper.setState(state);
    return wrapper;
  }

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

describe("<Footer />", () => {


  let wrapper;
  beforeEach(() => wrapper = shallow(<Footer />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => { 
      expect(wrapper.find('div').length).toEqual(6);
  });

  it('should render a <ul >', () => { 
    expect(wrapper.find('ul').length).toEqual(2);
  });

    it('should render a <li >', () => { 
        expect(wrapper.find('li').length).toEqual(4);
    });
 
})