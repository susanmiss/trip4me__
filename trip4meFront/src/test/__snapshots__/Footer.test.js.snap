// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Renders correctly 1`] = `ShallowWrapper {}`;
